types/jest/index.d.ts

```diff
@@ -727,7 +727,7 @@ declare namespace jest {
         not: T
     };
     // should be R extends void|Promise<void> but getting dtslint error
-    interface Matchers<R, T> {
+    interface Matchers<R, T = void> {
```

Comment:
It seems like `T` isn't used in return type positions, but `void` only really makes sense as the return type of a function.
If your intent is to have a default with no properties, could you use `{}` here instead? Thanks!

Reply:
Good catch @uniqueiniquity! 

Using `void` is likely to cause issues with `toMatchSnapshot`/`toMatchInlineSnapshot` when `T` remains unspecified. Looks like we're expecting an indexed type here:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c640483291a20a1e7d3ebe1f13a444c697d7be4c/types/jest/index.d.ts#L977

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c640483291a20a1e7d3ebe1f13a444c697d7be4c/types/jest/index.d.ts#L988

Reply:
it was gonna be `= {}` but @orta suggested `= void` instead - maybe he has some thoughts he'd like to share?

Reply:
Hah, yeah, I recommended both - but I figured `void` would be closer to your average expect-y function type. Seems fine to me to use `{}` instead.